backend/bitboxbase/bitboxbase.go

```diff
@@ -228,44 +263,220 @@ func (base *BitBoxBase) RPCClient() *rpcclient.RPCClient {
 // MiddlewareInfo returns the received MiddlewareInfo packet from the rpcClient
 func (base *BitBoxBase) MiddlewareInfo() (rpcmessages.SampleInfoResponse, error) {
 	if !base.active {
-		err := errp.New("Attempted a call to non-active base")
-		return rpcmessages.SampleInfoResponse{}, err
+		return rpcmessages.SampleInfoResponse{}, errp.New("Attempted a call to non-active base")
 	}
 	return base.rpcClient.GetSampleInfo()
 }
 
 // VerificationProgress returns the received VerificationProgress packet from the rpcClient
 func (base *BitBoxBase) VerificationProgress() (rpcmessages.VerificationProgressResponse, error) {
 	if !base.active {
-		err := errp.New("Attempted a call to non-active base")
-		return rpcmessages.VerificationProgressResponse{}, err
+		return rpcmessages.VerificationProgressResponse{}, errp.New("Attempted a call to non-active base")
 	}
 	return base.rpcClient.GetVerificationProgress()
 }
 
-// SyncWithOption returns true if the chosen sync option was executed successfully
-func (base *BitBoxBase) SyncWithOption(option SyncOption) (bool, error) {
+// ReindexBitcoin returns true if the chosen sync option was executed successfully
+func (base *BitBoxBase) ReindexBitcoin() error {
+	if !base.active {
+		return errp.New("Attempted a call to non-active base")
+	}
+	base.log.Println("bitboxbase is making a ReindexBitcoin call")
+	reply, err := base.rpcClient.ReindexBitcoin()
+	base.changeStatus(bitboxbasestatus.StatusInitialized)
+	if err != nil {
+		base.log.WithError(err)
+		// if the rpc client throws an error we don't want to pass that on to the frontend, because its probably junk
+		return nil
+	}
+	if !reply.Success {
+		return errp.Newf("ReindexBitcoin was not successful. Message: '%s'. Code: '%s'", reply.Message, reply.Code)
```

Comment:
if you slap an `Error() string` method on reply.Error, you can return that as a normal Go error, so the handler can inspect the error and relay the error code and message.

Like this, the frontend has no access to the error code.